hazelcast-code-generator/src/main/java/com/hazelcast/client/protocol/generator/CodeGenerationUtils.java

```diff
@@ -251,18 +280,34 @@ public static String getCSharpType(String type) {
 
             Iterator<String> iterator = typeParameters.iterator();
             while (iterator.hasNext()) {
-                builder.append(getCSharpType(iterator.next()));
+                builder.append(getLanguageType(iterator.next(), languageMapping));
                 if (iterator.hasNext()) {
-                    builder.append(',');
+                    builder.append(", ");
                 }
             }
-            builder.append(">");
-            return builder.toString();
+
+            if (type.startsWith("java.util.Map<")) {
```

Comment:
@cangencer OK, I got you point now. Updated the code so that the change is only for CPP.